src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ConstructedEETypeNode.cs

```diff
@@ -168,6 +166,8 @@ protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFact
             // Ask the metadata manager if we have any dependencies due to reflectability.
             factory.MetadataManager.GetDependenciesDueToReflectability(ref dependencyList, factory, _type);
 
+            CodeBasedDependencyAlgorithm.AddInterestingInteropConstructedTypeDependencies(ref dependencyList, factory, _type);
```

Comment:
Just want to double check that you're aware of this: for valuetypes this code path will only trigger if the type was boxed by the user code. I'm not sure if that should imply we want to be able to marshal it.